nixos/modules/system/activation/top-level.nix

```diff
@@ -306,6 +313,26 @@ in
       '';
     };
 
+    system.includeBuildDependencies = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Whether to include the build closure of the whole system in
+        its runtime closure.  This can be useful for making changes
+        fully offline, as it includes all sources, patches, and
+        intermediate outputs required to build all the derivations
+        that the system depends on.
+
+        Note that this includes _all_ the derivations, down from the
+        included applications to their sources, the compilers used to
+        build them, and even the bootstrap compiler used to compile
+        the compilers. This increases the size and build/download time
+        of the system drastically: a minimal configuration with no
+        extra services enabled grows from ~670MiB in size to 13.5GiB,
+        and takes proportionally longer to build.
```

Comment:
I don't think the build is much slower, is it? Isn't this more about substituting many more dependencies?